scripts/scr_interact.gml

```diff
@@ -0,0 +1,74 @@
+//Had to use step event for this script, can't find a way to detect leaving a collision otherwise
+
+//Check for collisions with interactable objects
+colobject = collision_rectangle(x,y,x+64,y+128,obj_interactable,0,1);
+
+//If the player is colliding with a door and they are not busy, execute the following code
+if(colobject and not global.transitioning)
+{
+    //Destroy last instance of the interaction button
+    if(instance_exists(obj_tempinteract))
+    {
+        with(obj_tempinteract)
+            instance_destroy();
+    }
+    
+    //Create an interaction button above the player
+    instance_create(x, y - 96, obj_tempinteract);
```

Comment:
Why are we recreating the instance every step? This is very expensive. May be easier to update `obj_tempinteract`'s  position instead of destroying it.